cpp/tests/copying/gather_str_tests.cpp

```diff
@@ -124,8 +125,8 @@ TEST_F(GatherTestStr, GatherDontCheckOutOfBounds)
                                       rmm::mr::get_current_device_resource());
 
   std::vector<char const*> h_expected;
-  for (auto itr = h_map.begin(); itr != h_map.end(); ++itr) {
-    h_expected.push_back(h_strings[*itr]);
+  for (int& itr : h_map) {
```

Comment:
do we need `int&` ref here ? I think `int` or `const int` should be sufficient. 
There is an inconsistency in using range based for loop with this and the above loop i.e in `TEST_F(GatherTestStr, Gather)`

Reply:
Good call, yes this is unnecessary. I'll remove the ref qualifier.